preload/prefetch-headers.html

```diff
@@ -0,0 +1,27 @@
+<!DOCTYPE html>
+<title>Ensures that prefetch sends headers as per-spec</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/common/utils.js"></script>
+<script src="resources/prefetch-helper.js"></script>
+<body>
+<script>
+
+promise_test(async t => {
+    const {href} = await prefetch({"type": "image/png", file: "../../images/green.png"}, t);
+    const [info] = await get_prefetch_info(href);
+    const {headers} = info;
+    assert_equals(headers["sec-fetch-dest"], "empty");
+    assert_equals(headers["sec-purpose"], "prefetch");
+    assert_equals(headers["Origin"], undefined);
+}, "Prefetch should include Sec-Purpose=prefetch and Sec-Fetch-Dest=empty headers");
```

Comment:
Can we have separate tests for the absence of the proprietary Purpose / X-moz / X-prefetch headers? (Separate so that people still get credit for passing this one, even if they fail that one.)

Reply:
You mean tests that pass or fail when those proprietary headers are present?

Reply:
Tests that fail when the spec isn't followed, i.e., when the headers are present.